core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -1338,15 +1342,35 @@ private void deleteSubjectCompatibility(String subject) throws StoreException {
 
   public Mode getMode(String subject) throws SchemaRegistryStoreException {
     try {
-      return lookupCache.mode(subject, false, defaultMode);
+      SchemaRegistryValue globalModeValue = this.kafkaStore.get(new ModeKey(null));
+      Mode globalMode = globalModeValue != null ? ((ModeValue) globalModeValue).getMode() : null;
```

Comment:
I think this can be replaced with `Mode globalMode = lookupCache.mode(null, false, defaultMode)`